app_integrations/apps/slack.py

```diff
@@ -0,0 +1,277 @@
+"""
+Copyright 2018-present, Airbnb Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import re
+import time
+
+from app_integrations import LOGGER
+from app_integrations.apps.app_base import StreamAlertApp, AppIntegration
+
+class SlackApp(AppIntegration):
+    """SlackApp will collect 2 types of event logs: access logs and integration logs.
+
+    This base class will be inherited by different subclasses based on different
+    event types.
+
+    Access logs:
+        contain information about logins
+
+    Integration logs:
+        contain details about your workspace's integrated apps
+    """
+
+    _SLACK_API_BASE_URL = 'https://slack.com/api/'
+    _SLACK_API_MAX_ENTRY_COUNT = 1000
+    _SLACK_API_MAX_PAGE_COUNT = 100
+
+    def __init__(self, config):
+        super(SlackApp, self).__init__(config)
+        self._next_page = 1
+
+    @classmethod
+    def _endpoint(cls):
+        """Class method to return the endpoint to be used for this slack instance
+
+        Returns:
+            str: Path of the desired endpoint to query
+
+        Raises:
+            NotImplementedError: If the subclasses do not properly implement this method
+        """
+        raise NotImplementedError('Subclasses should implement the _endpoint method')
+
+    @classmethod
+    def _type(cls):
+        raise NotImplementedError('Subclasses should implement the _type method')
+
+    @classmethod
+    def _sleep_seconds(cls):
+        raise NotImplementedError('Subclasses should implement the _sleep_seconds method')
+
+    @classmethod
+    def service(cls):
+        return 'slack'
+
+    @classmethod
+    def date_formatter(cls):
```

Comment:
If not special formatting is needed, you should be able to omit this method altogether (it's not an abstractmethod that is required to be implemented)